analyzers/src/SonarAnalyzer.CSharp/Rules/ClearTextProtocolsAreSensitive.cs

```diff
@@ -38,20 +38,19 @@ public sealed class ClearTextProtocolsAreSensitive : HotspotDiagnosticAnalyzer
         private const string DiagnosticId = "S5332";
         private const string MessageFormat = "Using {0} protocol is insecure. Use {1} instead.";
         private const string EnableSslMessage = "EnableSsl should be set to true.";
-        private const string HttpPattern = @"^http:\/\/(?!localhost|127.0.0.1).+";
-        private const string FtpPattern = @"^ftp:\/\/.*@(?!localhost|127.0.0.1)";
-        private const string Telnet = "telnet";
-        private const string TelnetPattern = @"^telnet:\/\/.*@(?!localhost|127.0.0.1)";
-        private const string TelnetPatternForIdentifier = @"(Telnet)(?![a-z])";
-        private const string LocalHost = "localhost";
-        private const string LocalHostIp = "127.0.0.1";
+
+        private const string TelnetKey = "telnet";
         private const string EnableSslName = "EnableSsl";
+        private const string ValidServerPattern = "localhost|127.0.0.1|::1";
 
-        private static readonly DiagnosticDescriptor DefaultRule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager).WithNotConfigurable();
+        private static readonly DiagnosticDescriptor DefaultRule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager).WithNotConfigurable();
+        private static readonly DiagnosticDescriptor EnableSslRule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, EnableSslMessage, RspecStrings.ResourceManager).WithNotConfigurable();
 
-        private static readonly DiagnosticDescriptor EnableSslRule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, EnableSslMessage, RspecStrings.ResourceManager).WithNotConfigurable();
+        private readonly Regex httpRegex = new Regex(@$"^http:\/\/(?!{ValidServerPattern}).+", RegexOptions.Singleline | RegexOptions.Compiled);
```

Comment:
nitpick: `new Regex(pattern, RegexOptions.Singleline | RegexOptions.Compiled);` can be a `CompiledRegex(string pattern)` private static method